### ISSUE TITLE
Is there a way to query json API by a specific version attribute...

### ISSUE THREAD

**Issue by pconesa** [pypa/packaging-problems#356]
Let me explain the case.

1.- We have a main python app that accepts plugins.
2.- Our app has a "CORE_VERSION" that the pypi packages (plugins) needs to mach
3.- We have a "Plugin manager" that queries: https://warehouse.pypa.io/api-reference/json/#get--pypi--project_name--json but needs to filter by its "CORE_VERSION".

With this when we release a new version or our app, that breaks the compatibility we can bump our "CORE_VERSION" and query/offer our users only those packages compatible with the new CORE_VERSION.

So far we are doing it using the "comment_text" whic is the only field we think appears in https://warehouse.pypa.io/api-reference/json/#get--pypi--project_name--json per release.

Please, note that we are aware o the relase entry point: https://warehouse.pypa.io/api-reference/json/#release but this degrade aour performance we will need to query, first the project entry point and then each of the release to find the last compatible one matching our CORE_VERSION.

**What is the problem of the "comment_text"?**... it is specified at "deploying time" (we use twine) as a parameter. If our users forgets the comment, it means is compatible with the oldest CORE_VERSION. I know, this was a bad decision on our side that I already regret.

One possible solution would be to make twine take the comment form the setup, not sure if this is possible. Another would be to be able **to have a field in the setup that is listed in the project API per release.**

Maybe the solution is out there already and I'm not seeing it. Sorry if  so.

All the best, Pablo.

**Comment by di** [comment]
Hi @pconesa, I moved this to pypa/packaging-problems as it seems this is a broad issue that might touch several different tools.

I'm not sure I entirely understand your use case though. You're setting some `CORE_VERSION` in the `comment` field, and you're querying `/pypi/<project_name>/json` to get it for every release, correct? Am I correct in assuming that the value you put in the `comment` field is not related to the actual version of the package? Can you give us some examples?

**Comment by pconesa** [comment]
Thanks @di , You got it right.

Let me explain with an example: 

Project url: https://pypi.org/pypi/scipion-em-cryosparc2/json

There, per release, the only field we've found we can get and we can set freeley is the comment:

So we have:
````
   "1.0.0b0": [
      {
        "comment_text": "scipion-2.0",

    "3.0.1": [
      {
        "comment_text": "scipion-3.0",
````
So version:
1.0.0.b0 --> scipion-2.0
3.0.1 --> scipion-3.0

We though about using major version to match the core but we wasn't sure how our version scheme was going to evolve, so we decided to decouple it.

**Comment by di** [comment]
How are these packages getting installed, is it with pip? If so, one thing that you could have done instead is create a dummy `scipion` package, which matches your core version, and then have each plugin declare a dependency on it, e.g. `scipion<3.0`. So when you install, you install both the plugin and the dummy package with it's version pinned, e.g.

```
pip intsall scipion-em-cryosparc2 scipion==2.0
```

and pip's resolver would find the compatible release automatically.

I'm guessing you're somewhat tied to the `comment` field at this point though, since you don't want to break backwards compatibility, is that correct?

**Comment by pconesa** [comment]
Yep, that is correct, our installation process is pip based but is hidden from the users. Let me try to explain:

1.- Users install our framework (we acually have a installer that creates a virtual env/conda) and installs the core packages (all pip based). At this point our software is useless without plugins. 

2.- User launches our "plugin manager" that needs to ask pypi for available compatible plugins. Here, to be able to "offer" plugins is where we need to do the matching using, now "comment_text" content with "framework" core version.

I think we can define dependencies properly to enforce versions compatibility (not 100% sure)...but this is not the issue. The key here is "offer".
